fe/fe-core/src/main/java/com/starrocks/load/InsertOverwriteJobRunner.java

```diff
@@ -0,0 +1,340 @@
+                job.setNewPartitionNames(info.getNewPartitionsName());
+                job.setJobState(InsertOverwriteJobState.OVERWRITE_PREPARED);
+                break;
+            case OVERWRITE_FAILED:
+                job.setJobState(InsertOverwriteJobState.OVERWRITE_FAILED);
+                LOG.info("replay insert overwrite job:{} to FAILED", job.getJobId());
+                gc();
+                break;
+            case OVERWRITE_CANCELLED:
+                job.setJobState(InsertOverwriteJobState.OVERWRITE_CANCELLED);
+                LOG.info("replay insert overwrite job:{} to CANCELLED", job.getJobId());
+                gc();
+                break;
+            case OVERWRITE_SUCCESS:
+                job.setJobState(InsertOverwriteJobState.OVERWRITE_SUCCESS);
+                doCommit();
+                LOG.info("replay insert overwrite job:{} to SUCCESS", job.getJobId());
+                break;
+            default:
+                LOG.warn("invalid to state:{} for insert overwrite job:{}", info.getToState(), job.getJobId());
+                break;
+        }
+    }
+
+    private void transferTo(InsertOverwriteJobState state) throws Exception {
+        if (state == InsertOverwriteJobState.OVERWRITE_SUCCESS) {
+            Preconditions.checkState(job.getJobState() == InsertOverwriteJobState.OVERWRITE_PREPARED);
+        }
+        InsertOverwriteStateChangeInfo info =
+                new InsertOverwriteStateChangeInfo(job.getJobId(), job.getJobState(), state,
+                        job.getSourcePartitionNames(), job.getNewPartitionNames());
+        GlobalStateMgr.getCurrentState().getEditLog().logInsertOverwriteStateChange(info);
+        job.setJobState(state);
+        handle();
+    }
+
+    private void prepare() throws Exception {
+        Preconditions.checkState(job.getJobState() == InsertOverwriteJobState.OVERWRITE_PENDING);
+        this.watershedTxnId =
+                GlobalStateMgr.getCurrentGlobalTransactionMgr().getTransactionIDGenerator().getNextTransactionId();
+        List<Partition> sourcePartitions = job.getOriginalTargetPartitionIds().stream()
+                .map(id -> targetTable.getPartition(id)).collect(Collectors.toList());
+        List<String> sourcePartitionNames = sourcePartitions.stream().map(p -> p.getName()).collect(Collectors.toList());
+        job.setSourcePartitionNames(sourcePartitionNames);
+        job.setNewPartitionNames(sourcePartitionNames.stream().map(name -> name + postfix).collect(Collectors.toList()));
+        transferTo(InsertOverwriteJobState.OVERWRITE_PREPARED);
+    }
+
+    private void executeInsert() throws Exception {
+        long insertStartTimestamp = System.currentTimeMillis();
+        insertStmt.setOverwrite(false);
+        // should replan here because prepareInsert has changed the targetPartitionNames of insertStmt
+        ExecPlan newPlan = new StatementPlanner().plan(insertStmt, context);
+        stmtExecutor.handleDMLStmt(newPlan, insertStmt);
```

Comment:
should we write audit.log after execute SQL?

Reply:
sql has been added to audit.log